### ISSUE TITLE
Locker-references borked?

### ISSUE THREAD

**Issue by ZodiacShadow** [SpadesNeil/VOREstation#610]
When trying to point (as in, with the point-verb) at either locker in the HoP's office, the connecting door to the northern hallway (between the office and the Conference Room) is pointed to, instead. This happened for both me and Joey.

**Comment by Arokha** [comment]
I can replicate this, but have NO IDEA how it's possible. It does the same thing for the HoS locker. But... NOT the CMO or Captain's locker, which are very similar paths.

**Comment by ZodiacShadow** [comment]
I will be so very interested in hearing this phenomenon explained.

**Comment by Arokha** [comment]
The Point To procs are so simple I can only think this is a BYOND bug. I can't see how this is possible with the code. It just makes a little pointy sprite and says the message. There's no witchcraft.

**Comment by ZodiacShadow** [comment]
And yet the witchcraft happens. BYOND can be so delightfully silly at times.
You mentioned HoS, CMO and Captain lockers.
What about the CE and RD lockers?
What's the common thread, here?

**Comment by Arokha** [comment]
None that I can see. Even just the CMO, HOS, and HOP being different is enough to break any commonality. They're all defined in the same file with the same code. None of them have special code. I'm compiling the game in BYOND 510.1339 to see if it has this issue.

**Comment by Arokha** [comment]
(Even if they did have special code it doesn't matter, the point-to verb doesn't actually do anything with the objects, only checks what turf they are on and points at it. Even then, it just says [src] points at [atom] for the message. The game thinks the locker IS the door, which isn't possible)

**Comment by ZodiacShadow** [comment]
...hah.
Can't help but wonder where in the process that order gets fucked up.

**Comment by Arokha** [comment]
This has to be a BYOND bug. I don't even know what.

Yes it does it to the CE's locker. 

I took the HOP's locker, dragged it across the room so it was next to the wall safe. Pointed at, still points at door.

Dragged it into the door space, still points at door.

Dragged it 1 tile north of door, points at locker. 

Dragged back into room next to wall safe again, points at locker. Ok so room is cursed?

Leave locker there, walk to 1 tile north of bottom door. Point at locker, points at locker. Ok so I'm cursed?

Walk back up to where I was standing before when pointing at locker, points at locker. Ok it's fixed?

Put locker back towards left side of room, point at locker, points at door.

**_?????????????????????????????????????????_**

**Comment by Arokha** [comment]
I should also mention if you cannot see the door, the pointing always works.

**Comment by ForFoxSake** [comment]
The names conflict.
Steps to reproduce: Debug game, point at locker, rename either locker or door and point again.
I believe the ' symbol is messing with the HTML escaping, remove it, everything works. Could be an issue in the point-to verb code.

**Comment by Arokha** [comment]
Even if items have the same name, it shouldn’t cause THIS, though. I think it’s more likely that renaming it pokes something else that causes it to stop freaking out, like moving it around does. If it’s the name, why does physically moving the locker change the symptoms?

**Comment by ForFoxSake** [comment]
Because the full names aren't being checked, just the "Head of Personnel" bit, moving the locker is changing the order in which the items are indexed in game, making one take priority over the other.

It's also worth noting that "head of personnel's locker" is the longest locker name out of any of them.

**Comment by Arokha** [comment]
The indexing order makes some sense except also YOU moving changes the symptoms. So maybe it’s objects relative to you.

But regardless, look at mob.dm and the pointed verb. It doesn’t use the name for anything.

**Comment by Arokha** [comment]
For example, standing next to the paper tray in the HOS office and pointing to the locker points to the locker. Standing 1 tile north points to the door.

**Comment by ForFoxSake** [comment]
`atom/A as mob|obj|turf in view()`
It is using the name in that bit of code, but it's resolving it on that line. This issue also occurs if two mobs have the same name and you use the admin jump-to-mob command.
(I suspect this is the reason monkeys were numbered.)

"In view" suggests it's using vision code, which works radially from the source object, north first, I'm guessing.

**Comment by Arokha** [comment]
They don't explain how view() works in the DM guide unfortunately since that's a stock byond proc.

I would imagine that view would return the objects though, and not the names, however the 'in' operator is virtually undocumented as well unfortunately. Does 'in' just compare the names of the referenced objects, instead of the IDs? Or are lists of objects or vars in reference to objects actually somehow a reference to the name? Or is the name used as some sort of 'primary key' that is returned when you use some unspecified comparison operator?

I mean I know list["Thingname"] is the way to reference items in lists but I'd always assumed that behind the scenes it was assuming I meant the 'name' var and was then iterating through the list to find it, rather than the list literally being names. Is that not true?

**Comment by Arokha** [comment]
Did more poking.
Points to locker: http://pastebin.com/G4g5vVui
Points to door: http://pastebin.com/dnyqakj0

That's the output of view() when standing in those two spots. The order is a bit strange but seems to go the left first, then ... well, it's kind of hard to follow. This is me guessing on the order after reading the lists but it's hard to tell where the buttons are since I'm too lazy to look at locs right now (have company over), and "floor" doesn't mean much.

View1: http://i.imgur.com/rMmQgYK.png
View2: http://i.imgur.com/8JO0Alo.png

Now I'm more interested in how this actually works rather than just 'rename thing to fix it'.

**Comment by ZodiacShadow** [comment]
You are really diligent about this. I like that.
I have yet to even have a cursory understanding of BYOND code, so I'm afraid I can't provide much input beyond sparking the investigation in the first place.
